### PR TITLE
chore: bump version to 10.1.0 w/ changelog

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+| bug fix |  **a11y:** list key manager not emitting change event if new item is added to active index ([#19666](https://github.com/angular/components/issues/19666)) ([3fbebe0](https://github.com/angular/components/commit/3fbebe0)), closes [#19661](https://github.com/angular/components/issues/19661) |
+| bug fix |  **aria-describer:** messages not being read out in IE and Edge ([#12304](https://github.com/angular/components/issues/12304)) ([0e22019](https://github.com/angular/components/commit/0e22019)), closes [#12298](https://github.com/angular/components/issues/12298) |
+| bug fix |  **autocomplete:** don't close when clicking inside custom origin ([#19784](https://github.com/angular/components/issues/19784)) ([1319c0a](https://github.com/angular/components/commit/1319c0a)), closes [#19771](https://github.com/angular/components/issues/19771) |
+| bug fix |  **benchmark:** fix license header and correct test descriptions ([#19596](https://github.com/angular/components/issues/19596)) ([0d266e2](https://github.com/angular/components/commit/0d266e2)) |
```
Comment (crisbeto) [inline CHANGELOG.md:22]: I don't think these `benchmark` commits are supposed to show up here.

Comment (andrewseguin) [reply CHANGELOG.md:22]: Do we want `benchmark` to show up at all in the CHANGELOG?

Comment (crisbeto) [reply CHANGELOG.md:22]: I don't think so, we probably should've marked these commits as `build`.

Comment (andrewseguin) [reply CHANGELOG.md]: SGTM - removed

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,105 @@
...
+| feature |  **tree:** add trackBy to NestedTreeControl ([#19602](https://github.com/angular/components/issues/19602)) ([e2c7e1a](https://github.com/angular/components/commit/e2c7e1a)) |
+
+### material
+
+|            |                       |
+| ---------- | --------------------- |
+| bug fix |  **a11y:** focus trap directive not capturing focus if auto capture input is set after init ([#19689](https://github.com/angular/components/issues/19689)) ([1acc608](https://github.com/angular/components/commit/1acc608)), closes [#19664](https://github.com/angular/components/issues/19664) |
+| bug fix |  **a11y:** list key manager not emitting change event if new item is added to active index ([#19666](https://github.com/angular/components/issues/19666)) ([3fbebe0](https://github.com/angular/components/commit/3fbebe0)), closes [#19661](https://github.com/angular/components/issues/19661) |
+| bug fix |  **aria-describer:** messages not being read out in IE and Edge ([#12304](https://github.com/angular/components/issues/12304)) ([0e22019](https://github.com/angular/components/commit/0e22019)), closes [#12298](https://github.com/angular/components/issues/12298) |
```
Comment (Splaktar) [inline CHANGELOG.md:20]: These 3 (this one and the 2 above) should be in the CDK section and not in `material`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,105 @@
...
+| bug fix |  **drag-drop:** not starting auto scroll when inside boundary ([#19865](https://github.com/angular/components/issues/19865)) ([39d48c7](https://github.com/angular/components/commit/39d48c7)), closes [#18596](https://github.com/angular/components/issues/18596) |
+| bug fix |  **drag-drop:** placeholder element not available in started event ([#19493](https://github.com/angular/components/issues/19493)) ([bc88cc1](https://github.com/angular/components/commit/bc88cc1)), closes [#19457](https://github.com/angular/components/issues/19457) |
+| bug fix |  **drag-drop:** sometimes incorrectly swapping items at the ends of the list ([#19633](https://github.com/angular/components/issues/19633)) ([0a6e4d6](https://github.com/angular/components/commit/0a6e4d6)), closes [#19249](https://github.com/angular/components/issues/19249) |
+| bug fix |  **drag-drop:** update free drag position on scroll ([#19648](https://github.com/angular/components/issues/19648)) ([9d903e0](https://github.com/angular/components/commit/9d903e0)) |
```
Comment (Splaktar) [inline CHANGELOG.md:34]: All of these `drag-drop` issues should be listed in `cdk` rather than `material`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,105 @@
...
+| bug fix |  **list:** single selection list selected state not shown on touch devices ([#19877](https://github.com/angular/components/issues/19877)) ([2eff545](https://github.com/angular/components/commit/2eff545)), closes [#19876](https://github.com/angular/components/issues/19876) |
+| bug fix |  **list:** use flexbox to size mat-line elements ([#19809](https://github.com/angular/components/issues/19809)) ([#19819](https://github.com/angular/components/issues/19819)) ([998bb18](https://github.com/angular/components/commit/998bb18)) |
+| bug fix |  **ng-update:** properly handle update from different working directory ([05ec25f](https://github.com/angular/components/commit/05ec25f)), closes [#19779](https://github.com/angular/components/issues/19779) |
+| bug fix |  **overlay:** outside click dispatcher not accounting for shadow DOM ([#19904](https://github.com/angular/components/issues/19904)) ([7427b26](https://github.com/angular/components/commit/7427b26)) |
+| bug fix |  **overlay:** overlay directives not emitting when detached externally ([#7950](https://github.com/angular/components/issues/7950)) ([4a3c960](https://github.com/angular/components/commit/4a3c960)) |
```
Comment (Splaktar) [inline CHANGELOG.md:44]: These 2 overlay issues should be in the `cdk` section.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,105 @@
...
+| bug fix |  focus monitor-based styles not working in some cases inside shadow dom ([#19422](https://github.com/angular/components/issues/19422)) ([df981ee](https://github.com/angular/components/commit/df981ee)), closes [#19414](https://github.com/angular/components/issues/19414) |
+| bug fix |  **testing:** harness selector not matching mat-spinner selector ([#19657](https://github.com/angular/components/issues/19657)) ([3ffcc63](https://github.com/angular/components/commit/3ffcc63)), closes [#19649](https://github.com/angular/components/issues/19649) |
+| feature |  **a11y:** Add optional home/end key support to ListKeyManager ([#19834](https://github.com/angular/components/issues/19834)) ([544e335](https://github.com/angular/components/commit/544e335)) |
+| feature |  **overlay:** support closing when clicking outside the overlay ([#16611](https://github.com/angular/components/issues/16611)) ([210d054](https://github.com/angular/components/commit/210d054)) |
```
Comment (Splaktar) [inline CHANGELOG.md:53]: The overlay and a11y features here should be in the `cdk` section.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,105 @@
...
+| feature |  **schematics:** enable routing option ([#19439](https://github.com/angular/components/issues/19439)) ([31723b8](https://github.com/angular/components/commit/31723b8)) |
+| feature |  **testing:** expose whether harness elements are focused ([#19704](https://github.com/angular/components/issues/19704)) ([081d833](https://github.com/angular/components/commit/081d833)), closes [#19702](https://github.com/angular/components/issues/19702) |
+| feature |  **testing:** Add a test harness for `MatToolbar` ([#19976](https://github.com/angular/components/issues/19976)) ([9e97a34](https://github.com/angular/components/commit/9e97a34)) |
+| feature |  **testing:** Add a test harness for `MatCard` ([#19833](https://github.com/angular/components/issues/19833)) ([54c2d16](https://github.com/angular/components/commit/54c2d16)) |
+| performance |  **int-bar:** Convert MatInkBarFoundation adapter to class object ([#19986](https://github.com/angular/components/issues/19986)) ([78dc50b](https://github.com/angular/components/commit/78dc50b)) |
```
Comment (Splaktar) [inline CHANGELOG.md:58]: `int-bar` -> `ink-bar`

**Comment by Splaktar** [comment]
Opened PR https://github.com/angular/components/pull/20045 to fix these issues.

**Pull request merged**